File: Darts/src/burlton/dartzee/code/screen/PlayerSummaryPanel.java
package burlton.dartzee.code.screen;

import burlton.dartzee.code.db.GameEntity;
import burlton.dartzee.code.db.PlayerEntity;
import burlton.dartzee.code.screen.stats.player.PlayerStatisticsScreen;
import burlton.dartzee.code.stats.PlayerSummaryStats;
import net.miginfocom.swing.MigLayout;

import javax.swing.*;
import javax.swing.border.TitledBorder;
import java.awt.*;
import java.awt.event.ActionEvent;
import java.awt.event.ActionListener;

public class PlayerSummaryPanel extends JPanel
								implements ActionListener
{
	private PlayerEntity player = null;
	private int gameType = -1;
	
	public PlayerSummaryPanel(int gameType)
	{
		this.gameType = gameType;
		
		setBorder(new TitledBorder(null, GameEntity.getTypeDesc(gameType), TitledBorder.LEADING, TitledBorder.TOP, new Font("Tahoma", Font.PLAIN, 20)));
		setLayout(new MigLayout("", "[][][][][][][][][grow][]", "[][][]"));
		
		add(lblP, "cell 0 0");
		
		add(lblW, "cell 2 0,alignx leading");
		
		add(lblHighScore, "cell 4 0");
		nfGamesPlayed.setEditable(false);
		add(nfGamesPlayed, "cell 0 1,growx");
		nfGamesPlayed.setColumns(10);
		Component horizontalStrut = Box.createHorizontalStrut(20);
		add(horizontalStrut, "cell 1 1");
		nfGamesWon.setEditable(false);
		add(nfGamesWon, "cell 2 1,growx");
		nfGamesWon.setColumns(10);
		
		Component horizontalStrut_1 = Box.createHorizontalStrut(20);
		add(horizontalStrut_1, "cell 3 1");
		nfBestGame.setEditable(false);
		add(nfBestGame, "cell 4 1,growx");
		nfBestGame.setColumns(10);
		
		Component horizontalStrut_2 = Box.createHorizontalStrut(20);
		add(horizontalStrut_2, "flowx,cell 8 1");
		btnViewStats.setFont(new Font("Tahoma", Font.PLAIN, 16));
		add(btnViewStats, "cell 8 1,alignx center");
		
		btnViewStats.addActionListener(this);
	}
	
	private final JTextField nfGamesPlayed = new JTextField();
	private final JTextField nfGamesWon = new JTextField();
	private final JTextField nfBestGame = new JTextField();
	private final JButton btnViewStats = new JButton("View Stats");
	private final JLabel lblP = new JLabel("Played");
	private final JLabel lblW = new JLabel("Won");
	private final JLabel lblHighScore = new JLabel("High score");
	
	public void init(PlayerEntity player)
	{
		setVisible(true);
		
		this.player = player;
		
		PlayerSummaryStats stats = PlayerSummaryStats.getSummaryStats(player, gameType);
		
		int gamesPlayed = stats.getGamesPlayed();
		int gamesWon = stats.getGamesWon();
		int bestScore = stats.getBestScore();
		
		nfGamesPlayed.setText("" + gamesPlayed);
		nfGamesWon.setText("" + gamesWon);
		
		if (bestScore > 0)
		{
			nfBestGame.setText("" + bestScore);
		}
		else
		{
			nfBestGame.setText("");
		}
		
		btnViewStats.setEnabled(gamesPlayed > 0);
	}
	
	
	@Override
	public void actionPerformed(ActionEvent arg0)
	{
		if (arg0.getSource() == btnViewStats)
		{
			PlayerStatisticsScreen statsScrn = ScreenCache.getScreen(PlayerStatisticsScreen.class);
			statsScrn.setVariables(gameType, player);
			
			ScreenCache.switchScreen(statsScrn);
		}
	}
}

File: src/main/kotlin/dartzee/screen/PlayerSummaryPanel.kt
package dartzee.screen

import dartzee.db.PlayerEntity
import dartzee.screen.stats.player.PlayerStatisticsScreen
import dartzee.stats.PlayerSummaryStats
import dartzee.utils.getTypeDesc
import net.miginfocom.swing.MigLayout
import java.awt.Font
import java.awt.event.ActionEvent
import java.awt.event.ActionListener
import javax.swing.*
import javax.swing.border.TitledBorder

class PlayerSummaryPanel(private val gameType: Int) : JPanel(), ActionListener
{
    private var player: PlayerEntity? = null

    private val nfGamesPlayed = JTextField()
    private val nfGamesWon = JTextField()
    private val nfBestGame = JTextField()
    private val btnViewStats = JButton("View Stats")
    private val lblP = JLabel("Played")
    private val lblW = JLabel("Won")
    private val lblHighScore = JLabel("High score")

    init
    {
        border = TitledBorder(null, getTypeDesc(gameType), TitledBorder.LEADING, TitledBorder.TOP, Font("Tahoma", Font.PLAIN, 20))
        layout = MigLayout("", "[][][][][][][][][grow][]", "[][][]")

        add(lblP, "cell 0 0")

        add(lblW, "cell 2 0,alignx leading")

        add(lblHighScore, "cell 4 0")
        nfGamesPlayed.isEditable = false
        add(nfGamesPlayed, "cell 0 1,growx")
        nfGamesPlayed.columns = 10
        val horizontalStrut = Box.createHorizontalStrut(20)
        add(horizontalStrut, "cell 1 1")
        nfGamesWon.isEditable = false
        add(nfGamesWon, "cell 2 1,growx")
        nfGamesWon.columns = 10

        val strutOne = Box.createHorizontalStrut(20)
        add(strutOne, "cell 3 1")
        nfBestGame.isEditable = false
        add(nfBestGame, "cell 4 1,growx")
        nfBestGame.columns = 10

        val strut2 = Box.createHorizontalStrut(20)
        add(strut2, "flowx,cell 8 1")
        btnViewStats.font = Font("Tahoma", Font.PLAIN, 16)
        add(btnViewStats, "cell 8 1,alignx center")

        btnViewStats.addActionListener(this)
    }

    fun init(player: PlayerEntity)
    {
        isVisible = true

        this.player = player

        val stats = PlayerSummaryStats.getSummaryStats(player, gameType)

        val gamesPlayed = stats.gamesPlayed
        val gamesWon = stats.gamesWon
        val bestScore = stats.bestScore

        nfGamesPlayed.text = "" + gamesPlayed
        nfGamesWon.text = "" + gamesWon

        if (bestScore > 0)
        {
            nfBestGame.text = "$bestScore"
        }
        else
        {
            nfBestGame.text = ""
        }

        btnViewStats.isEnabled = gamesPlayed > 0
    }


    override fun actionPerformed(arg0: ActionEvent)
    {
        if (arg0.source === btnViewStats)
        {
            val statsScrn = ScreenCache.getScreen(PlayerStatisticsScreen::class.java)
            statsScrn.setVariables(gameType, player!!)

            ScreenCache.switchScreen(statsScrn)
        }
    }
}
